StoreManagementService/src/PBJ.StoreManagementService.DataAccess/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.EntityFrameworkCore;
+using PBJ.StoreManagementService.DataAccess.Context;
+using PBJ.StoreManagementService.DataAccess.Entities.Abstract;
+using PBJ.StoreManagementService.DataAccess.Repositories.Abstract;
+
+namespace PBJ.StoreManagementService.DataAccess.Repositories
+{
+    public class BaseRepository<TEntity> : IRepository<TEntity>
+        where TEntity : BaseEntity
+    {
+        protected readonly DatabaseContext _databaseContext;
+
+        public BaseRepository(DatabaseContext databaseContext)
+        {
+            _databaseContext = databaseContext;
+        }
+
+        public virtual async Task<List<TEntity>> GetAmountAsync(int amount)
+        {
+            return await _databaseContext.Set<TEntity>().AsNoTracking()
```

Comment:
You can use global NoTracking in the Context class:
     ChangeTracker.QueryTrackingBehavior = QueryTrackingBehavior.NoTracking;

Most likely you will not need tracking functionality

Reply:
Ok, I will fix than in next prs